cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -472,6 +480,26 @@ pub(crate) fn define<'shared>(
             ),
     );
 
+    // XX /r with runtime REX emission based on instruction Type.
+    recipes.add_template_recipe(
+        EncodingRecipeBuilder::new("rr_runtime_rex_rb", &formats.binary, 1)
+            .operands_in(vec![gpr, gpr])
+            .operands_out(vec![0])
+            .compute_size("size_plus_maybe_rex_rb_for_two_in_regs")
+            .emit(
+                r#"
+                    let rex_byte = rex2(in_reg0, in_reg1);
+                    // The "put" functions assert whether the REX prefix is needed.
+                    if rex_byte == BASE_REX {
+                        put_op1(bits, rex_byte, sink);
+                    } else {
+                        put_rexop1(bits, rex_byte, sink);
```

Comment:
@sstangl What did you think of my previous suggestion in the above comment?

Reply:
In general, no, they cannot be unified. The issue is that there are some instructions that require an empty REX prefix, and so the comparison against BASE_REX is not valid for every kind of instruction. However, those instructions wouldn't use recipes like this because REX is non-inferrable.

We could however have a third kind of put function, that does the above `BASE_REX` comparison. That would move the code out of the `recipes.rs` at least, at the expense of complicating the templates, which need to be taught about this new kind of put function.

The follow-up patch I'm preparing to this one kind of does that.

Reply:
Yeah, third kind was what I had in mind ("a temporary put_ function"). Ok I'll wait for the follow-up then!